just_audio_background/lib/just_audio_background.dart

```diff
@@ -839,3 +859,42 @@ extension _ValueStreamExtension<T> on ValueStream<T> {
   /// Backwards compatible version of valueOrNull.
   T? get nvalue => hasValue ? value : null;
 }
+
+/// Notification config, used to set system notification properties.
+@immutable
+class NotificationConfig {
+  const NotificationConfig({
+    this.controls = const [],
+    this.systemActions = const {},
+    this.androidCompactActionIndices,
+  });
+
+  /// Not supported starting with Android 13 (API level 33)
+  /// 
+  /// More info: https://developer.android.com/about/versions/13/behavior-changes-13#playback-controls
+  final List<int>? androidCompactActionIndices;
+  final List<MediaControl> controls;
+  final Set<MediaAction> systemActions;
+}
+
+
+/// Snapshot of current player state.
+/// Used to provide data to [NotificationConfigBuilder].
+abstract class PlayerStateSnapshot {
+  /// -1 if no next index available
+  int get nextIndex;
+  /// -1 if no previous index available
+  int get previousIndex;
+  /// Whether player is currently in playing state.
+  bool get playing;
+  /// Whether player is currently paused.
+  bool get paused => !playing;
```

Comment:
I'd rather not have a redundant `paused` since it has no basis in just_audio's existing API design.

Reply:
I thought it would be convenient for end user to have paused getter, bit I'll remove it then, no prob